src/structure.jl

```diff
@@ -1,24 +1,32 @@
 """
-Compute the reciprocal lattice. Takes special care of 1D or 2D cases.
-We use the convention that the reciprocal lattice is the set of G vectors such
-that G ⋅ R ∈ 2π ℤ for all R in the lattice.
+Compute the inverse of the lattice. Takes special care of 1D or 2D cases.
 """
-function compute_recip_lattice(lattice::AbstractMatrix{T}) where {T}
+function compute_inverse_lattice(lattice::AbstractMatrix{T}) where {T}
     # Note: pinv pretty much does the same, but the implied SVD causes trouble
     #       with interval arithmetic and dual numbers, so we go for this version.
     n_dim = count(!iszero, eachcol(lattice))
     @assert 1 ≤ n_dim ≤ 3
     if n_dim == 3
-        2T(π) * inv(lattice')
+        inv(lattice')
     else
-        2T(π) * Mat3{T}([
+        Mat3{T}([
             inv(lattice[1:n_dim, 1:n_dim]')   zeros(T, n_dim, 3 - n_dim);
             zeros(T, 3 - n_dim, 3)
         ])
     end
 end
 
 
+"""
+Compute the reciprocal lattice.
+We use the convention that the reciprocal lattice is the set of G vectors such
+that G ⋅ R ∈ 2π ℤ for all R in the lattice.
+"""
+function compute_recip_lattice(lattice::AbstractMatrix{T}) where {T}
+    2T(π) * compute_inverse_lattice(lattice)
+end
```

Comment:
T not needed, could make this a one-liner.

Reply:
I am not sure how you write one-liners.
Btw, with the use of pseudo-inverse I do not seem to obtain the correct forces anymore. I will invistigate a bit more.

Reply:
Ah no sorry, what I had in mind does not work, because you need the `T`